gramex/handlers/mlhandler.py

```diff
@@ -261,14 +271,56 @@ def _transform(self, data, **kwargs):
         data = self._filterrows(data, **kwargs)
         return data
 
+    def _chooseScore(self, data, target):
+        try:
+            print( 'What score are we reading here? -----> ', self.score_pref)
+            print('What class category are we reading here? -----> ', self.multiclass)
+            predict = self.model.predict(data)
+            print('PREDICT ----->', predict)
+            print('TARGET ------>', target)
+
+            #FOR CLASSIFICATION
+            if(self.score_pref == 'f1-score'):
+                return f1_score(target, predict, average= 'weighted') # can change average value
+
+            elif(self.score_pref == 'accuracy'):
+                return accuracy_score(target, predict)
+
+
+            elif(self.score_pref == 'recall'):
+                return recall_score(target, predict, average=None)
+
+            elif(self.score_pref == 'roc_auc_score'):
+                result = roc_auc_score(target, predict)
+                if(self.multiclass == 'True'):
+                    return roc_auc_score(target, predict, multi_class='ovr')
+                return result 
+
+            #FOR REGRESSION    
+            #elif(self.score_pref == 'mean_squared_error'):
+            #   return mean_squared_error(target, predict)
+
+            elif(self.score_pref == 'r2'):
+                return r2_score(target, predict)
+
+            else:
+                return self.model.score(data,target)
+
+        except ValueError:
+            print('ValueError occured. Please check if the scoring metric is appropriate for the given data!')
+
+
+
     def _predict(self, data=None, score_col=''):
         if data is None:
             data = self._parse_data(False)
         data = self._transform(data, drop_duplicates=False)
         self.model = cache.open(self.model_path, joblib.load)
         try:
             target = data.pop(score_col)
-            return self.model.score(data, target)
+            s = self._chooseScore(data, target)
+            print('s: ',s)
+            return s
         except KeyError:
```

Comment:
Here, the user will be sending the scoring metric as a request parameter, named `_metric`. So we can use the snippet above to change this.